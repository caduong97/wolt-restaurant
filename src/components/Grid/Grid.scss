@import '../../assets/styles/utils';

$default-max-width: 1280px;
$max-columns: 10;

@mixin columns-num {
  @for $i from 1 through $max-columns {
    &--#{$i} {
      grid-template-columns: repeat($i, 1fr);

      @include for-tablet {
        grid-template-columns: repeat(
          auto-fit,
          minmax((992px - $spacing-04 * 2 - $spacing-01 * 2) / 3, 1fr)
        );
      }

      @include for-tablet-small {
        grid-template-columns: repeat(
          auto-fit,
          minmax((768px - $spacing-04 * 2 - $spacing-01) / 2, auto)
        );
      }

      @include for-mobile {
        grid-template-columns: repeat(auto-fit, minmax(100%, auto));
      }
    }
  }
}

.grid {
  width: 100%;
  display: grid;
  justify-items: stretch;
  gap: $spacing-02;

  @include columns-num;

  & > * {
    object-fit: cover;
    text-align: left;
    margin: 0 auto;
  }
}
