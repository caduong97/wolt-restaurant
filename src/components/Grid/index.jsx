import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './Grid.scss'

const Grid = ({ colNum, children }) => {
  const gridClasses = classNames('grid', `grid--${colNum}`)
  return <div className={gridClasses}>{children}</div>
}

Grid.propTypes = {
  colNum: PropTypes.number.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
}

export default Grid
