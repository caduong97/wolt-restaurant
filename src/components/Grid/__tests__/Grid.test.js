import React from 'react'
import { render } from '@testing-library/react'
import Grid from '..'

describe('<Grid/>', () => {
  it('renders and matches snapshot', () => {
    const { container } = render(
      <Grid colNum={1}>
        <p>Lorem Ipsum</p>
      </Grid>,
    )

    expect(container).not.toBeNull()
    expect(container).toMatchSnapshot()
  })

  it('append column prop to classname', () => {
    const colNum = 2
    const { container } = render(
      <Grid colNum={colNum}>
        <p>Lorem Ipsum</p>
        <p>Lorem Ipsum</p>
      </Grid>,
    )

    expect(container.firstChild).toHaveClass(`grid grid--${colNum}`)
  })
})
