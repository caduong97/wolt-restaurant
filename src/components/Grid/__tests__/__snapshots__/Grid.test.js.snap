// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid/> renders and matches snapshot 1`] = `
<div>
  <div
    class="grid grid--1"
  >
    <p>
      Lorem Ipsum
    </p>
  </div>
</div>
`;
